clients/src/main/java/org/apache/kafka/clients/ClientResponse.java

```diff
@@ -104,6 +135,50 @@ public long requestLatencyMs() {
         return latencyMs;
     }
 
+    public Logger getLogger() {
+        if (logContext == null) {
+            LogContext logContext = new LogContext("[" + requestHeader.toString() + "] ");
+            return logContext.logger(ClientResponse.class);
+        }
+        return logContext.logger(ClientResponse.class);
+    }
+
+    public void releaseBuffer() {
+        if (memoryPool != null && responsePayload != null) {
+            getLogger().trace(
+                "ByteBuffer[{}] returned to memorypool ({}). Ref Count: {}. RequestType: {}",
+                (responsePayload == null ? "null" : responsePayload.position()), (memoryPool == null ? "null" : "not null"),
+                refCount.get(), this.requestHeader.apiKey());
+            memoryPool.release(responsePayload);
+            responsePayload = null;
+        } else {
+            // Mostly this means double releaseBuffer calls or releaseBuffer where response is null such as disconnects
+            getLogger().error(
+                "ByteBuffer[{}] returned to memorypool ({}). Ref Count: {}. RequestType: {}",
+                (responsePayload == null ? "null" : responsePayload.position()), (memoryPool == null ? "null" : "not null"),
+                refCount.get(), this.requestHeader.apiKey());
+        }
+    }
+
+    public void checkBufferRelease() {
+        if (memoryPool != null && responsePayload != null) {
+            getLogger().error("ByteBuffer[{}] not released. Ref Count: {}. RequestType: {}",
+                responsePayload.position(), refCount.get(), this.requestHeader.apiKey());
+            memoryPool.release(responsePayload);
```

Comment:
The line above this says buffer is not released and we are releasing the buffer here. Am I missing something here?

Reply:
Provided some more clarification as part of [comment](https://github.com/tusharg1993/kafka/pull/2#discussion_r656650607). Let me know if it's still unclear.